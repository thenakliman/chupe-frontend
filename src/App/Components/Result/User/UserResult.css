.UserResult {
  width: 50%;
  margin: 1% 25%;
  display: flex;
}

.user-tile {
  width: 25%;
  height: 10rem;
  margin: 2%;
  border: 1px solid #eee;
}

.user-image {
  width: 100%;
  height: 8rem;
}