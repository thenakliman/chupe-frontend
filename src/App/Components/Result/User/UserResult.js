import propTypes from 'prop-types';
import React from 'react';

require('./UserResult.css');

/**
* User result component of the application.
*
* @author: thenakliman
*/
export class UserResult extends React.Component {
/** Fetch users from the backend and update in the store. */
  componentDidMount() {
    this.props.getUsers();
  }
/**
* User result component of the application.
*
* @return {Object} UserResult component.
*/
  render() {
    return (
        <div className='UserResult'>
          {
            this.props.users.map((user) =>(
                <div key={`${user.userName}`} className={'user-tile'}>
                    <img src='fake' className='user-image'/>
                    <div>{user.firstName} {user.lastName}</div>
                </div>))
          }
        </div>
    );
  }
}

UserResult.propTypes = {
  users: propTypes.arrayOf(propTypes.object).isRequired,
  getUsers: propTypes.func.isRequired,
};
