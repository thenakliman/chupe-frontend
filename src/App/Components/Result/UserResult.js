import propTypes from 'prop-types';
import React from 'react';

require('./UserResult.css');

/**
* User result component of the application.
*
* @author: thenakliman
*/
export class UserResult extends React.Component {
/**
* User result component of the application.
*
* @return {Object} UserResult component.
*/
  render() {
    return (
        <div className='UserResult'>
          <ul id='all-users-list'>
          {
            this.props.users.map((user) =>(
                <li key={`${user.userName}`}>
                    {user.firstName} {user.secondName} {user.email}
                </li>))
          }
          </ul>
        </div>
    );
  }
}

UserResult.propTypes = {
  users: propTypes.arrayOf(propTypes.object).isRequired,
};
