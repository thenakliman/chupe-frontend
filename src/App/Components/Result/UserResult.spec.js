import React from 'react'; // eslint-disable-line no-unused-vars
import {UserResult} from './UserResult'; // eslint-disable-line no-unused-vars
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('User result component snapshot', () => {
    it('should match the snapshot', () => {
        const users = [{'username': 'user1'}, {'username': 'user2'}];
        const wrapper = shallow(<UserResult users={users}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('User result component', () => {
    it('Display list of users', () => {
        const users = [
            {'username': 'user1'},
            {'username': 'user2'},
            {'username': 'user3'},
        ];
        const wrapper = shallow(<UserResult users = {users}/>);
        expect(wrapper.find('#all-users-list').get(0).props.children.length)
            .toEqual(3);
    });
});
