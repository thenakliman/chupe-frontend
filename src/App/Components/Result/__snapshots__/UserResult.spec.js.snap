// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User result component snapshot should match the snapshot 1`] = `
<div
  className="UserResult"
>
  <ul
    id="all-users-list"
  >
    <li
      key="undefined"
    >
       
       
    </li>
    <li
      key="undefined"
    >
       
       
    </li>
  </ul>
</div>
`;
